pipeline/eBOSS_QSO/QuasarMapper.py

```diff
@@ -0,0 +1,191 @@
+import matplotlib.pyplot as plt
+from astropy.io import fits
+from astropy.table import Table
+
+import os
+import pandas as pd
+import collections
+import numpy as np
+import healpy as hp
+
+import pymaster as nmt
+
+
+class QuasarMapper:
+    def __init__(self): 
+        #Inputs: the four fits files from eBOSS DR14
+        
+        data_path = '/home/zcapjru/PhD/Data/'

```

Comment:
Now, all of these hard-coded things you can replace by more generic looping over the input file paths:
```
self.cat_data = []
self.cat_random = []
for file_data, file_random in zip(data_list, random_list):
    if not os.path.isfile(file_data):
        raise ValueError(f"File {file_data} not found")
    with fits.open(file_data) as f:
        self.cat_data.append(Table.read(f).to_pandas()
    if not os.path.isfile(file_random):
        raise ValueError(f"File {file_random} not found")
    with fits.open(file_random) as f:
        self.cat_random.append(Table.read(f).to_pandas()
```
At this point, `self.cat_data` and `self.cat_random` will be lists of catalog tables containing data and randoms.